components/sup/src/main.rs

```diff
@@ -82,6 +83,17 @@ fn main() {
     logger::init();
     let mut ui = UI::default_with_env();
     let flags = FeatureFlag::from_env(&mut ui);
+    // On Windows initializng the signal handler will create a ctrl+c handler for the
+    // process which will disable default windows ctrl+c behavior and allow us to
+    // handle via check_for_signal. However, if the supervsor is in a long running
+    // non-run hook, the below loop will not get to check_for_signal in a reasonable
+    // amount of time and the supervisor will not respond to ctrl+c. On Windows, we
```

Comment:
initializng -> initializing supervsor -> supervisor

Reply:
I've fixed these typos but also tried to clarify this comment which I found a bit confusing because it didn't really reflect reality.